Caddy fails to renew certificate for admin interface

with the configuration at the bottom, caddy renews the internal certificate for the admin interface, but fails to reload it. instead the following error is logged:

```
panic: certificate worker: runtime error: invalid memory address or nil pointer dereference
goroutine 35 [running]:
github.com/caddyserver/certmagic.(*jobManager).worker.func1()
	github.com/caddyserver/certmagic@v0.17.2/async.go:58 +0x65
panic({0x18763c0, 0x2a9a9c0})
	runtime/panic.go:884 +0x212
github.com/caddyserver/certmagic.(*Cache).replaceCertificate(_, {{{0xc000348660, 0x2, 0x2}, {0x199a3e0, 0xc000348870}, {0x0, 0x0, 0x0}, {0x0, ...}, ...}, ...}, ...)
	github.com/caddyserver/certmagic@v0.17.2/cache.go:292 +0x31
github.com/caddyserver/certmagic.(*Config).reloadManagedCertificate(_, {_, _}, {{{0xc000348660, 0x2, 0x2}, {0x199a3e0, 0xc000348870}, {0x0, 0x0, ...}, ...}, ...})
	github.com/caddyserver/certmagic@v0.17.2/certificates.go:341 +0x418
github.com/caddyserver/certmagic.(*Cache).queueRenewalTask.func1()
	github.com/caddyserver/certmagic@v0.17.2/maintain.go:240 +0x510
github.com/caddyserver/certmagic.(*jobManager).worker(0x2aa5720)
	github.com/caddyserver/certmagic@v0.17.2/async.go:73 +0x11b
created by github.com/caddyserver/certmagic.(*jobManager).Submit
	github.com/caddyserver/certmagic@v0.17.2/async.go:50 +0x29b
```

not sure if this is a bug in caddy or certmagic but since it can be triggered with the following caddy configuration, created it as a caddy issue.
 
```json
{
  "admin": {
    "identity": {
      "identifiers": [
        "127.0.0.2"
      ],
      "issuers": [
        {
          "ca": "local",
          "lifetime": "3m",
          "module": "internal",
          "sign_with_root": false
        }
      ]
    },
    "remote": {
      "listen": "127.0.0.2:2021"
    }
  },
  "apps": {
    "http": {
      "servers": {}
    },
    "pki": {
      "certificate_authorities": {
        "local": {
          "install_trust": false,
          "intermediate_common_name": "Example Intermediate CA",
          "name": "Example CA",
          "root_common_name": "ca.example.bug"
        }
      }
    }
  },
  "logging": {
    "logs": {
      "default": {
        "level": "DEBUG"
      }
    }
  }
}
```

--- a/admin.go
+++ b/admin.go
@@ -572,12 +572,13 @@ func replaceRemoteAdminServer(ctx Context, cfg *Config) error {
 }
 
 func (ident *IdentityConfig) certmagicConfig(logger *zap.Logger, makeCache bool) *certmagic.Config {
+	var cmCfg *certmagic.Config
 	if ident == nil {
 		// user might not have configured identity; that's OK, we can still make a
 		// certmagic config, although it'll be mostly useless for remote management
 		ident = new(IdentityConfig)
 	}
-	cmCfg := &certmagic.Config{
+	template := certmagic.Config{
 		Storage: DefaultStorage, // do not act as part of a cluster (this is for the server's local identity)
 		Logger:  logger,
 		Issuers: ident.issuers,
@@ -589,7 +590,8 @@ func (ident *IdentityConfig) certmagicConfig(logger *zap.Logger, makeCache bool)
 			},
 		})
 	}
-	return certmagic.New(identityCertCache, *cmCfg)
+	cmCfg = certmagic.New(identityCertCache, template)
+	return cmCfg
 }
 
 // IdentityCredentials returns this instance's configured, managed identity credentials

